src/pages/home/report/ReportActionContextMenuItem.js

```diff
@@ -29,52 +35,107 @@ function getButtonState(isHovered = false, isPressed = false) {
 const propTypes = {
     icon: PropTypes.elementType.isRequired,
     text: PropTypes.string.isRequired,
+    successIcon: PropTypes.elementType,
+    successText: PropTypes.string,
     isMini: PropTypes.bool,
+    onPress: PropTypes.func.isRequired,
 };
 
 const defaultProps = {
     isMini: false,
+    successIcon: null,
+    successText: '',
 };
 
-const ReportActionContextMenuItem = (props) => {
-    const {getButtonStyle, getIconFillColor} = getReportActionContextMenuItemStyles(props.isMini);
-    return (
-        props.isMini
-            ? (
-                <Tooltip text={props.text}>
-                    <Pressable style={({hovered, pressed}) => getButtonStyle(getButtonState(hovered, pressed))}>
+class ReportActionContextMenuItem extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            hasRun: false,
+        };
+        this.onPress = this.onPress.bind(this);
+    }
+
+    /**
+     * Called on button press and mark the run
+     *
+     * @memberof ReportActionContextMenuItem
+     */
+    onPress() {
+        const pressResult = this.props.onPress();
+        if (!(pressResult instanceof Promise)) {
```

Comment:
Hmmm so NAB (really just thinking "out loud"), but generally speaking this code does something with the result of a function, but we're not sure if that function is synchronous or asynchronous. Maybe we could make a generic utility function to make this simpler. Something like this perhaps:

```javascript
function makeAsync(func) {
    try {
        const result = func();
        return result instanceof Promise ? result : Promise.resolve(result);
    } catch (e) {
        return Promise.reject(e);
    }
}
```

Then in this instance we would just do:

```javascript
const asyncOnPress = makeAsync(this.props.onPress);
asyncOnPress().then(result => this.setState({success: result}));
```

Reply:
where should I put this utility?

Reply:
So current use case does not require promise as return value. So for now, I will remove the promise check and the return value would `Boolean`.

Reply:
Hey, I _think_ we are overthinking this a little. I would suggest just making it a requirement to pass a `Promise` for this prop if you need to handle something async here.

Another question however... why must we await this promise anyway? 
Is this something that we expect to happen very slow or actually fail ever?
Can we just make it fire and forget?

Reply:
Yeah. Point taken.

Reply:
> Can we just make it fire and forget

I don't think so in this case, because after the promise resolves we're making a visual change to show that it happened successfully (and I don't think it makes sense to use Onyx in this case). But yeah we can just always pass an async function here, which should be no problem!

Sorry for leading you astray with on this @parasharrajat, my solution was just a bit over-engineered.

Reply:
Ok. Currently, Clipboard.setString() is a sync function thus I also think to keep it sync. I expect it to be async in the future so I thought of handling the Promise here. What do you say @roryabraham, should I keep the async behaviour here and return `Promise` from the onPress?

Reply:
Meh, you're first to implement one of the callbacks in this menu so just do the simplest thing that works for now and get rid of the unnecessary promise check. Sorry I was confused because I thought `Clipboard.setString()` was async. Anyways, the next person can worry about making sure it works for async functions too 🤪